backend/src/controllers/stockController.ts

```diff
@@ -135,4 +135,37 @@ export const updateStockItemQuantity = async (
         console.error('Error in updateStockItemQuantity:', err);
         res.status(500).json({error: 'Error while updating the database.'});
     }
-}
\ No newline at end of file
+}
+
+export const addStockItem = async (
+    res: Response,
+    connection: PoolConnection,
+    stockID: number,
+    item: { ID: number; LABEL: string;  DESCRIPTION: string; QUANTITY: number }
+) => {
+    try {
+        const {ID, LABEL, DESCRIPTION,QUANTITY} = item;
+        await connection.query("INSERT INTO items VALUES (?, ?, ?, ? ,?)", [
+            ID,
+            LABEL,
+            DESCRIPTION,
+            QUANTITY,
+            stockID
+        ]);
+
+        if (res && res.json) {
+            res.json({message: "Item created successfully."});
+        } else {
+            throw new Error(
+                "Response or res.json is undefined. Cannot call res.status and res.json for error handling."
+            );
+        }
+    } catch (err: any) {
+        console.error(err);
+        if (res && res.json) {
+            //TODO :affiner les message d'erreur.
```

Comment:
issue(blocking): copie d'un todo qui dit d'affiner le message d'erreur.

Reply:
fix in next PR

Reply:
fix in next PR